src/python/pants/bin/local_pants_runner.py

```diff
@@ -106,7 +106,8 @@ def _maybe_init_graph_session(graph_session, options_bootstrapper,build_config,
 
       v2_ui = options.for_global_scope().v2_ui
       zipkin_trace_v2 = options.for_scope('reporting').zipkin_trace_v2
-      graph_session = graph_scheduler_helper.new_session(zipkin_trace_v2, RunTracker.global_instance().run_id, v2_ui)
+      should_report_workunits = False
```

Comment:
I'm unsure on the exact purpose of this `should_report_workunits` variable that we pass around in the code. If we're only going to hardcode it here, should it really exist? If the idea is to eventually extend it into a user facing option, can we maybe make the most of `should_report_zipkin_spans` and either use it or rename it to something more appropriate if we're only concerned about its name?

I'm a bit confused, so feel free to explain what's on your mind as much as possible and don't assume I know exactly what I'm talking about :smile:

Reply:
OK. I read this from the PR's description:
```
Future commits that make use of this functionality will add logic to flip that boolean on/off as required, and define a callback function doing useful work that can be passed into AsyncWorkunitHandler.
```
That helps. Maybe I would recommend including the minimum amount of these commits to propagate `should_report_workunits` until a point where it becomes clear how it is used (if we still think it's necessary and not redundant with `should_report_zipkin_spans`). That would make this code easier to review.

If you are quite keen on merging this PR sooner rather than later, I could close an eye on this and continue this discussion on review of the next PR that adds these commits.

Reply:
Yeah, I had envisioned this specific PR to be a subset of the work that was easily reviewable, rather than the entirety of the feature. There's a tension between creating smaller PRs that are easier to review but that have lines of code that only make sense in a larger context, and creating gigantic, hard to review PRs that have everything relevant to a feature.

With respect to this `should_report_workunits` variable specifically, I'm deliberately avoiding changing any code that affects zipkin reporting as it currently exists, even if in the future it might make sense to do so, so I can avoid breaking currently-working functionality while iterating on this async workunits work. Eventually yes I agree it should probably replace `should_report_zipkin_spans` as the thing that triggers the rust code to report workunits to python or not.

Reply:
> There's a tension between creating smaller PRs that are easier to review but that have lines of code that only make sense in a larger context, and creating gigantic, hard to review PRs that have everything relevant to a feature.

Definitely not always easy to cut the right size. As a rule of thumb, I think it's nice to have some kind of atomicity where all the changes within one PR can be justified in the context of other changes in the same PR. The reason is that if you don't have that kind of atomicity, it's easy to introduce semi-unused code that may get forgotten about and remain dormant for a long time. (It's much easier to avoid introducing code than it is to remove it after the fact).

It's not always feasible, though and I'm OK with this PR as is for now given your reasons. Will just try to keep this discussion in mind when reviewing the follow ups.